src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -209,10 +209,46 @@ public SNIHandle CreateConnectionHandle(object callbackObject, string fullServer
                 case TdsEnums.TCP:
                     return ConstructTcpHandle(serverNameParts[1], timerExpire, callbackObject, parallel);
 
+                case TdsEnums.NP:
+                    if (parallel)
+                    {
+                        SNICommon.ReportSNIError(SNIProviders.INVALID_PROV, 0, SNICommon.MultiSubnetFailoverWithNonTcpProtocol, string.Empty);
+                        return null;
+                    }
+
+                    // Named Pipe Format: np:\\<host name>\pipe\<pipe name>
+                    const int minPipeLength = 10; // eg: //./pipe/a
+                    string pipePath = serverNameParts[1].Trim();
+                    if (pipePath.Length < minPipeLength)
+                    {
+                        SNILoadHandle.SingletonInstance.LastError = new SNIError(SNIProviders.NP_PROV, 0, SNICommon.InvalidConnStringError, string.Empty);
+                        return null;
+                    }
+
+                    if (pipePath[0] != '\\' || pipePath[1] != '\\')
+                    {
+                        SNILoadHandle.SingletonInstance.LastError = new SNIError(SNIProviders.NP_PROV, 0, SNICommon.InvalidConnStringError, string.Empty);
+                        return null;
+                    }
+
+                    int endofServerName = pipePath.IndexOf('\\', 2);
+
+                    string pipeToken = @"\pipe\";
```

Comment:
Have you considered using System.Uri for getting the pipe related information? 
If System.Uri can handle the parsing of the pipe path, without adding any additional project dependencies, it would be better.